### DISCUSSION TITLE
Basic question about GET request parameters and SqlModel/Pydantic

### DISCUSSION THREAD

**Discussion by blcoyote** [fastapi/fastapi#3781] [Questions] [answered]
Hi,
I have a basic question that I've been unable to find an answer to in the documentation - or I missed it.

I have a weather station that is sending data to an api I've got running in my lan.
it's encoding all its data as parameters in the url and sending the lot of it via a GET for some dumb reason - including id and password.

I'm extracting the values and inserting it into a SqlModel.
I would like to clean up the code and I'm wondering if fastapi can parameters into an instance of a model automatically - since the model and the parameter values are named identically:

![cropped](https://user-images.githubusercontent.com/30732216/131227062-b42c9047-f613-4e30-b474-9d50256e42bf.jpeg)

Ideally I would like a mapping like when you're passing a body in a POST which then maps the parameters as an instance of a model that I can then pass to the database - without having to map everything manually.

TLDR: How do I get rid of the lines of code marked in red?

**Comment by ghandic** [comment, marked as answer]
If you dont mind changing it from the query parameters to the body of the HTTP request you can just use a pydantic model - eg your `Observation` model. However, if you require it to be in the query parameters then you will need to use a `Depends` to use the dependency injection system to add in the common parameters and return your object (note you could utilise unpacking here as shown below)


```python
from fastapi import FastAPI, APIRouter, Depends
from pydantic import BaseModel

app = FastAPI(docs_url="/")
router = APIRouter()


class CreateUser(BaseModel):
    username: str
    password: str


def create_user(username: str, password: str):
    return CreateUser(**locals())


@router.get("/weather-query")
def login(login: CreateUser = Depends(create_user)):
    # login is now a CreateUser as it comes from the dependency injection system
    return login


@router.get("/weather-body")
def login(login: CreateUser):
    # login is now a CreateUser by default as it comes from the request body
    return login


app.include_router(router=router, prefix="/example")
```

**Reply by ghost** [reply]
I didn't even think of depends. I will try this. thanks.

**Reply by ghandic** [reply]
@BriLeaChr could you mark this as answered if it helped and works? Thanks 😄

**Reply by ghost** [reply]
I will. I just got to test when i'm not at work. :)

**Reply by ghost** [reply]
I think is will work with SQLModel too as its based on Pydantic

**Reply by ghandic** [reply]
Should do!

**Reply by blcoyote** [reply]
Managed to test it and sign in with the right account. Thanks it works. also with SQLModels without any adaption
